draft-johani-dnsop-multi-provider-requirements-00.md

```diff
@@ -148,6 +148,199 @@ to be able to fully support all multi-provider scenarios:
+
+(More scenarios are needed)
+
+## Simple multi-signer Scenario
+
+This subsection analyses the simple multi-signer scenario
+
+### Zone contents
+
+Lets assume that each signer has a KSK and a ZSK called KSK_A, ZSK_A, KSK_B,
+and ZSK_B. Each provider has NS records, NS_A for provider A and NS_B for
+provider B.
+
+The quiescent state is the following:
+
+Provider A signs the zone (except for the DNSKEY, CDS and CDNSKEY RRsets) with
+ZSK_A. Provider B signs with ZSK_B. Provider A's DNSKEY RRset contains at 
+least KSK_A, ZSK_A, and ZSK_B. And is signed by at least KSK_A. Similarly,
+provider B's DNSKEY RRset contains at least KSK_B, ZSK_A, and ZSK_B. This
+RRset is signed by KSK_B. Both providers have an NS RRset that consists of
+NS_A and NS_B. At the parent the DS records refer to KSK_A and KSK_B.
+
+### ZSK rolls
+
+RFC 6781 describes two ZSK rolls, a Pre-Publish ZSK roll and a
+Double-Signature ZSK roll. In a multi-signer environment, the Double-Signature
+ZSK roll has fewer synchronization requirements, but is generally undesirable
+because it leads to more signatures. We will first look at the Double-Signature
+ZSK roll.
+
+We assume that the zone of provide A is currently signed with ZSK_A and
+that provider wants to roll to ZSK_An. Provider A starts by signing the
+zone with ZSK_An in addition to ZSK_A. This is a local operation of
+provider A. Provider A waits until all nameservers of provider A serve the
+new zone and any cache have expired. 
+
+Now provider A changes its DNSKEY RRset to consist of KSK_A, ZSK_An and ZSK_B
+and informs provider B to replace ZSK_A with ZSK_An. Provider A monitors
+the propagation of its own zone. Provider A also need to get confirmation
+from provider B that ZSK_An is fully propagated.
+
+Once that state is reached, provider A stops signing with ZSK_A and the key
+roll is complete.
+
+For the Pre-Publish ZSK roll, provider A first adds ZSK_An to its DNSKEY RRset
+and informs provider B that is now has two ZSKs: ZSK_A and ZSK_An.
+Provider A monitors the propagation of its own zone and waits for confirmation
+from provider B that the new DNSKEY RRset has fully propagated.
+
+Then provider A starts signing with ZSK_An and monitors propagation. This is
+a local operation.
+
+Finally, provider A removes ZSK_A from the DNSKEY RRset and informs provider B
+that it should only include ZSK_An in its DNSKEY RRset. The key roll is now
+complete though provider A SHOULD monitor the propagation of its DNSKEY
+RRset and ask provider B to report on propagation as well.
+
+### KSK rolls
+
+RFC 6781 defines two KSK rolls: the Double-Signature KSK roll and the
+Double-DS KSK roll. The Double-Signature KSK roll is easiest and we will
+look that first.
+
+We assume provider A wants to roll to KSK_An. First provider A adds
+KSK_An to its DNSKEY RRset. Provider A monitors the propagation of its
+DNSKEY RRset.
+
+When that has happened, provider A and B need to agree on new CDS/CDNSKEY
```

Comment:
Not sure what "agree" means. It's always the union of the CDS/CDNSKEY RRsets which the providers would have if they were the only one (just like for NS). "Agreeing" sounds like a bidirectional negotiation, but what's needed is only a unidirectional way of importing stuff from the other provider (and waking up to do it at the right time).